HW5(OOP_vs_FP)/HW5_Trupkina.js

```diff
@@ -0,0 +1,236 @@
+    { name: 'Thomas A. Anderson', age: 58 },
+    { name: 'Millie Brown', age: 17 },
+    { name: 'Harry Potter', age: 31 }
+];
+function findAdults(arr) {
+    const arrOver18 = [];
+    for (let i = 0; i < arr.length; i++) {
+        if (arr[i].age >= 18) {
+            arrOver18.push(arr[i]);
+        }
+    }
+    return arrOver18;
+}
+console.log(findAdults(arrPeople));
+/*(4) [{…}, {…}, {…}, {…}]
+0: {name: "Pedro Pascal", age: 45}
+1: {name: "Leonardo Dicaprio", age: 46}
+2: {name: "Thomas A. Anderson", age: 58}
+3: {name: "Harry Potter", age: 31} */
+
+//2.2 hof
+function findAdults_hof(arr, fn) {
+    const arrOver18 = [];
+    for (let i = 0; i < arr.length; i++) {
+        if (fn(arr[i]) === undefined) i++;
+        arrOver18.push(fn(arr[i]));
+    }
+    return arrOver18;
+}
+function more18(elem) {
+    if (elem.age >= 18) return elem.name;
+}
+console.log(findAdults_hof(arrPeople, more18));
+//(4) ["Pedro Pascal", "Leonardo Dicaprio", "Thomas A. Anderson", "Harry Potter"]
+
+//2.2.1 hof filter
+const findAdults_filter = arrPeople.filter(arrPeople => arrPeople.age >= 18);
+console.log(findAdults_filter);
+/*(4) [{…}, {…}, {…}, {…}]
+0: {name: "Pedro Pascal", age: 45}
+1: {name: "Leonardo Dicaprio", age: 46}
+2: {name: "Thomas A. Anderson", age: 58}
+3: {name: "Harry Potter", age: 31}*/
+
+
+// 3.
+//3.1
+const arrNum = [34, 38, 45, 9, 23];
+function findSumArrNumb(numb) {
+    let sum = 0;
+    for (let i = 0; i < numb.length; i++) {
+        sum += numb[i];
+    }
+    return sum;
+}
+console.log(findSumArrNumb(arrNum));//149
+
+
+//3.2.1 reduce________________________________
+const sum = arrNum.reduce(function (accumulator, currentValue) { //.reduce(function(accum, currentValue, index[0...n], array) {
+    return accumulator + currentValue;
+});
+console.log(sum);//25
+
+//   4.
+const arrPeopleName = ['Pedro', 'Leonardo', 'Anderson', 'Millie', 'Harry'];
+
+function findNameLength_hof(arr, fn) {
+    const arrlen = [];
+    for (let i = 0; i < arr.length; i++) {
+        arrlen.push(fn(arr[i]));
+    }
+    return arrlen;
+}
+function retName(item) {
+    return item.length;
+}
+const arrMap = findNameLength_hof(arrPeopleName, retName);
+console.log(arrMap);
+
+
+//  ______________Task for working with OOP:_______________
+// ES6
+class Company {
+
+    constructor(options) {
+        this.name = options.name;
+        this.vacation = options.vacation;
+    }
+    startVacation() { }
+    finishVacation() { }
+}
+// Inheritance class
+class PersonnelDep extends Company {
+
+    startVacation() {//Polymorphism shows which version of method should be run
+        this.vacation = true;
+        console.log(`I'm on vacation! :D`) //
+    }
+    finishVacation() {
+        this.vacation = false;
+        console.log(`I'm at work...`);
+    }
+
+    get getAdvice() {//encapsulation
+        let getInspirit = 'you were inspired...'
+        return console.log(getInspirit);//you were inspired...
+
+    }
+}
```

Comment:
what happened when we try get "new PersonnelDep()" ?

you work with this.name & this.vacation - but I do not see them